### PR TITLE
[W-11062526] RD Next Donation

### PR DESCRIPTION
**Pull request by maurimart**
[Technical Design Document W-11062526](https://salesforce.quip.com/qDizAf2L5ZdM)

# Critical Changes
Add fields : 
npe03__Total_Paid_Installments__c,
npe03__Installments__c,
npe03__Next_Payment_Date__c,

Add custom label : RD2_nextDonation

# Changes
Change next donation rendering method after getting table data.

### CODE REVIEW

File: force-app/main/default/lwc/rd2RecurringDonation/rd2RecurringDonation.js [hunk 1]
```diff
@@ -309,14 +308,11 @@ export default class RecurringDonationTable extends LightningElement {
       retrieveTableView()
         .then((data) => {
           if (data) {
+            console.log(JSON.parse(JSON.stringify(data)));
```
Comment (JoseRossiSF) [inline force-app/main/default/lwc/rd2RecurringDonation/rd2RecurringDonation.js:311]: Shouldn't this console log be removed?

**Review by gerardorodriguezsf — APPROVED** [review]
LGTM

**Review by force2b — CHANGES_REQUESTED** [review]
Please address merge conflicts

File: force-app/main/default/labels/CustomLabels.labels-meta.xml [hunk 2]
```diff
@@ -1697,6 +1697,14 @@
         <shortDescription>Error for DIs with populated RD fields when RD2 is disabled.</shortDescription>
         <value>You must enable Enhanced Recurring Donations before you can process Recurring Donations.</value>
     </labels>
+    <labels>
+        <fullName>RD2_nextDonation</fullName>
+        <categories>Recurring Donation</categories>
+        <language>en_US</language>
+        <protected>true</protected>
+        <shortDescription>Label for next donation mapping.</shortDescription>
+        <value>Paid Installments: {0}/{1},&lt;/br&gt; Next Installment Date: {2}.</value>
```
Comment (force2b) [inline force-app/main/default/labels/CustomLabels.labels-meta.xml:1706]: I recommend this as two labels (one of which is probably already the label of a field). I also recommend not using merge fields nor including the `</br>` tag. This can be rendered in the UI as two separate fields, such as:
```
<label.RD2_PaidInstallments>: value
<label.npe03__Recurring_Donation__c.NextDonationDate.label>: value
```
That will greatly help with translations and formatting.

**Review by force2b — CHANGES_REQUESTED** [review]

File: force-app/main/default/labels/CustomLabels.labels-meta.xml (hunk 2 above)
Comment (maurimart) [reply force-app/main/default/labels/CustomLabels.labels-meta.xml]: I appreciate the suggestion! the br has already been deleted, since for this case there is a logic that allows modifying the label from the admin and that this be reflected directly in the component in a dynamic way taking into account the line breaks, the design requirements and the ellipses of the paragraphs

File: force-app/main/default/labels/CustomLabels.labels-meta.xml [hunk 3]
```diff
@@ -1697,6 +1697,15 @@
...
         <value>You must enable Enhanced Recurring Donations before you can process Recurring Donations.</value>
     </labels>
+    <labels>
+        <fullName>RD2_nextDonation</fullName>
+        <categories>Recurring Donation</categories>
+        <language>en_US</language>
+        <protected>true</protected>
+        <shortDescription>Label for next donation mapping.</shortDescription>
+        <value>Paid Installments: {0}/{1},
+Next Installment Date: {2}.</value>
```
Comment (force2b) [inline force-app/main/default/labels/CustomLabels.labels-meta.xml:1707]: I would still prefer two separate labels. Best practice is that we do not escape labels when displaying. Generally embedded a CRLF into the label may not render as expected.

**Review by force2b — CHANGES_REQUESTED** [review]

File: force-app/main/default/labels/CustomLabels.labels-meta.xml (hunk 3 above)
Comment (maurimart) [reply force-app/main/default/labels/CustomLabels.labels-meta.xml]: changes applied!

File: force-app/main/default/lwc/rd2RecurringDonation/rd2RecurringDonation.html [hunk 4]
```diff
@@ -156,17 +156,14 @@
...
+                            <div class="next-donation" lwc:dom="manual"></div>
                         </td>
                         <td key={item.lastModified} data-id={item.recurringDonation.Id} class={tdClasses}  data-label={labels.lastModified}>
-                            <div class={rowClasses} style={fixedWidth} title="Last Modified">
+                            <div class={rowClasses} style={fixedWidth} title={labels.lastModified}>
                                 {item.lastModified}</div>
                         </td>
-                        <td data-label="Actions">
-                            <lightning-button-menu class="dv-dynamic-mobile" menu-alignment="auto" alternative-text="Action menu">
+                        <td data-label={labels.RD2_Actions}>
```
Comment (force2b) [inline force-app/main/default/lwc/rd2RecurringDonation/rd2RecurringDonation.html:165]: `data-label` is not UI text. It's strictly an internal html dom attribute and does not require (and probably should not be) dynamic text.

**Review by force2b — CHANGES_REQUESTED** [review]
This has a [build failure](https://metaci.sfdc.sh/builds/644103) due to a class compilation issue

**Comment by force2b** [comment]
@maurimart Have you looked at the build failure issue yet? https://metaci.sfdc.sh/builds/644103

File: force-app/main/default/lwc/rd2RecurringDonation/rd2RecurringDonation.html (hunk 4 above)
Comment (alvaroolivenciasf) [reply force-app/main/default/lwc/rd2RecurringDonation/rd2RecurringDonation.html:165]: @lparrott is it possible you continue with Michael’s change request and review Mauricio's reply ? Michael is on PTO during the next week and we would appreciate to finish with this PR during the first days of the next week.
FYI @gerardorodriguezsf

Comment (lparrott) [reply force-app/main/default/lwc/rd2RecurringDonation/rd2RecurringDonation.html:165]: If this Label ("Actions") actually shows up in the UI, then what you did is fine.

File: force-app/main/default/lwc/rd2RecurringDonation/rd2RecurringDonation.js [hunk 5]
```diff
@@ -359,6 +355,18 @@ export default class RecurringDonationTable extends LightningElement {
                     return { actions, ...el, nexDonationFormatFirstElement, nexDonationFormatSecondElement };
                 });
             }
+        }).finally(() => {
+          this.data?.forEach((el) => {
```
Comment (lparrott) [inline force-app/main/default/lwc/rd2RecurringDonation/rd2RecurringDonation.js:359]: This is pretty hard to follow, why was the pattern changed from using attributes on each 'item' in 'data'? Couldn't you follow similar logic when setting `this.data = data.map((el)...` above?

**Review by lparrott — CHANGES_REQUESTED** [review]

**Review by lparrott — APPROVED** [review]
